api/pkg/apis/projectcalico/v3/globalnetworkpolicy.go

```diff
@@ -115,6 +115,20 @@ type GlobalNetworkPolicySpec struct {
 
 	// NamespaceSelector is an optional field for an expression used to select a pod based on namespaces.
 	NamespaceSelector string `json:"namespaceSelector,omitempty" validate:"selector"`
+
+	// ProgramIntoDataplane, if set to OnDemand (the default), Felix will
+	// determine whether the policy is active on the local node automatically
+	// (i.e. when it applies to a local endpoint).  If set to Always, Felix
+	// will treat the policy as active (and program it into the dataplane) even
+	// if it is not used on the local node.
+	//
+	// In almost all cases, OnDemand is the best setting.  Always should only
+	// be used for static/baseline policies that are very likely to apply on
+	// every node (and it is only worthwhile if you have thousands of rules in such
+	// policies).  Using it on those policies prevents Felix from doing work
+	// to add/remove the policies when the first/last endpoint is added/removed.
+	// This reduces latency to add/remove that first/last endpoint.
+	ProgramIntoDataplane string `json:"programIntoDataplane,omitempty" validate:"omitempty,oneof=OnDemand Always"`
```

Comment:
The other option I thought of was applying hysteresis: once programmed, keep a policy programmed for X minutes even if it becomes unused.  But that impacts everyone, not just the folks who need this feature, and it makes the calculation graph stateful, which makes it far harder to test.  It could be managed but I much prefer the simple/explicit option, I think!

In practice, the large static policies are created before any workloads so we can't really tell if they're going to apply to "almost everything" or not until the workloads show up.